agent/utils/dockerapi_compare_versions_test.go

```diff
@@ -0,0 +1,135 @@
+// +build unit
+// Copyright 2014-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package utils
+
+import (
+	"testing"
+
+)
+
+func TestParseDockerVersions(t *testing.T) {
+	testCases := []struct {
+		input  string
+		output dockerapi_version
+	}{
+		{
+			input: "1.0",
+			output: dockerapi_version{
+				major: 1,
+				minor: 0,
+			},
+		},
+		{
+			input: "5.2",
+			output: dockerapi_version{
+				major: 5,
+				minor: 2,
+			},
+		},
+	}
+
+	for i, testCase := range testCases {
+		result, err := parseDockerVersions(testCase.input)
+		if err != nil {
```

Comment:
nit: can you use the [assert](https://godoc.org/github.com/pgpst/pgpst/internal/github.com/stretchr/testify/assert) package (e.g. assert.Nil in this case)? same for other places in this file. i think we normally use assert instead of if block + t.Error in the test.